docs/scripts/md_to_html.py

```diff
@@ -1,84 +1,116 @@
+"""Build documentation from Markdown files."""
+from pathlib import Path, PurePosixPath
+import sys
+from typing import TYPE_CHECKING
+
 import markdown
 
+if TYPE_CHECKING:
+    from typing import Union
+
+
+NON_DOCS_FOLDERS = ["assets", "css", "scripts"]
 
-########## RENDER HTML TOP ################
 
 def render_html_top() -> str:
+    """Render the top of the HTML page."""
+    top_html = """<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="UTF-8">
+        <title>BattINFO</title>
+        <link rel="stylesheet" type="text/css" href="./css/style.css">
+        <link rel="icon" type="image/x-icon" href="./assets/favicon.ico">
+    </head>
+<body>
+"""
+
+    banner = """
+<div class="banner">
+    <a href="index.html">
+        <img src="./assets/banner.jpg" alt="Banner Image">
+    </a>
+</div>
+"""
 
-    top_html = """
-        <!DOCTYPE html>
-        <html>
-            <head>
-                <meta charset="UTF-8">
-                <title>BattINFO</title>
-                <link rel="stylesheet" type="text/css" href="./css/style.css">
-                <link rel="icon" type="image/x-icon" href="./assets/favicon.ico">
-        </head>  
-        <body>      
-        """
-
-    banner =  '''
-        <div class="banner">
-            <a href="index.html">
-                <img src="./assets/banner.jpg" alt="Banner Image">
-            </a>
-        </div>
-        '''
-    
     return top_html + banner
 
-########## RENDER HTML BOTTOM ################
 
 def render_html_bottom() -> str:
+    """Render the bottom of the HTML page."""
     return """
-            </body>
-        </html>
-        """
+    </body>
+</html>
+"""
 
 
-########## LOAD MD INTO HTML  ################
+def load_md_into_html(path: "Union[str, Path]") -> str:
+    """Load Markdown file and convert to HTML."""
+    # Read Markdown file
+    markdown_text = path.read_text(encoding="utf-8")
 
-def load_md_into_html(path:str)-> str:
-    
-    with open(path, 'r', encoding="utf-8") as file:
-        markdown_text = file.read()
+    # Convert (and return) Markdown to HTML
+    return markdown.markdown(markdown_text)
 
-    # Convert Markdown to HTML
-    html = markdown.markdown(markdown_text)
-    return html
 
+def rendering_workflow(
+    output_path: Path = Path("site"),
+) -> None:
+    """Workflow for rendering the HTML pages."""
+    repo_dir = Path(__file__).resolve().parent.parent.parent.resolve()
+    docs_dir = repo_dir / "docs"
 
+    if not output_path.is_absolute():
+        # Relative path - Must be relative to the repository root
+        output_path = (repo_dir / output_path).resolve()
 
-########### RUN THE RENDERING WORKFLOW ##############
-
-def rendering_workflow():
+    # Get all Markdown files NOT in the assets, css, or scripts directories
+    md_files = [
```

Comment:
I dont fully follow the code from lines 63 to 85. I think you are scanning through md files not in the assets, css and scripts, but you have already moved them there. What is the purpose of this part of the code?